cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl_dpc.cpp

```diff
@@ -233,6 +233,37 @@ void train_kernel_hist_impl<Float, Bin, Index, Task>::init_params(train_context_
         bin_borders_host_[clmn_idx] = ind_ftrs.get_bin_borders(clmn_idx).to_host(queue_);
     }
 
+    // Copy bin borders to the device
+    const Index max_bins = desc.get_max_bins();
+    bin_borders_device_ =
+        pr::ndarray<Float, 1>::empty(queue_, { ctx.column_count_ * max_bins }, alloc::device);
+
+    auto bin_borders_ptr = bin_borders_device_.get_mutable_data();
+    const sycl::range<1> range = de::integral_cast<std::size_t>(ctx.column_count_);
+    // Copying to the device is not convenient,
+    // because original bin borders are std::vector<ndarray<Float, 1>>
+    auto bin_borders_ptrs_host = pr::ndarray<const Float*, 1>::empty(queue_, { ctx.column_count_ });
```

Comment:
```suggestion
    auto bin_borders_ptrs_host = pr::ndarray<Float*, 1>::empty(queue_, { ctx.column_count_ });
```

1) You'll change values here
2) Generally speaking shared memory usage is not recommended